File: javascript/parensMatch.js
const { assert } = require('./test-utils');

/**
 * @param {string} str  the string to validate
 * @returns {boolean} whether or not the given string has only matching parentheses
 */
function parensMatch(str) {}

assert(parensMatch(''), true, 'empty string validates');
assert(parensMatch('['), false, 'single bracket doesnt match');
assert(parensMatch('[()]'), true, '[()] works');
assert(parensMatch('[()[]]'), true, '[()[]] works');
assert(parensMatch('[()[]]]'), true, '[()[]]] doesnt work');

File: python/parensMatch.py
'''
@param {string} str  the string to validate
@returns {boolean} whether or not the given string has only matching parentheses
 '''
def parensMatch(candidate):
    return False

assert parensMatch(''), 'empty string validates'
assert not parensMatch('['), 'single bracket doesnt match'
assert parensMatch('[()]'), '[()] works'
assert parensMatch('[()[]]'), '[()[]] works'
assert not parensMatch('[()[]]]'), '[()[]]] doesnt work'
